sync/tryrwmutex.go

```diff
@@ -0,0 +1,103 @@
+package sync
+
+import (
+	"runtime"
+	"sync"
+	"sync/atomic"
+)
+
+const (
+	readOffset  = uint64(1)
+	writeOffset = uint64(1 << 20)
+	tryOffset   = uint64(1 << 40)
+
+	tryMask = uint64(1099511627775) // equivalent to setting the first 39 bits to '1'.
```

Comment:
Wow I don't even remember writing this code.

Also, seems like our Sync stuff should be pulled out into a separate package. Or maybe each sync tool pulled out separately.

I bet a lot of go devs would like these things.

We should also pull `Flush` from the TG, I think we've given up on using that. It's a nice concept but in terms of practical engineering it's just not worthwhile. Plus Flush is the whole reason we can't do nested Add calls, which otherwise I think would be supported.